spinalcordtoolbox/scripts/sct_run_batch.py

```diff
@@ -175,6 +180,35 @@ def _find_nonsys32_bash_exe():
     return shutil.which('bash', path=nonsys32_paths)
 
 
+def _parse_dataset_directory(path_data, subject_prefix="sub-", ignore_ses=False):
+    """
+    Parse a dataset directory to find subject directories (or session subdirectories, if present).
+
+    Notes:
+        - The dataset is assumed to be structured in a (somewhat) BIDS-compliant way, see:
+          https://bids-specification.readthedocs.io/en/stable/02-common-principles.html#file-name-structure
+        - This function is a rudimentary version of the library PyBIDS: https://github.com/bids-standard/pybids.
+          TODO: https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/3415
+    """
+    # Find subject directories within the root dataset folder
+    subject_dirs = sorted([f for f in os.listdir(path_data)
```

Comment:
A drive-by suggestion: I noticed in the docs for `listdir()` that [`scandir()`](https://docs.python.org/3/library/os.html#os.scandir) is a good option when you also want file attributes (like `is_dir()`) while iterating through filenames. Also, the `os.DirEntry` objects it returns conveniently have both `name` (the base filename) and `path` (the full path including the filename) as attributes.

However, you can't sort a list of `os.DirEntry` objects without specifying a key, for example:
```py3
subject_dirs = [d for d in os.scandir(path_data)
                if d.name.startswith(subject_prefix)
                and d.is_dir()]
subject_dirs.sort(key=lambda d: d.name)
```
(Alternatively, wait until the end of the function to sort the list of full paths.)

Reply:
> However, you can't sort a list of os.DirEntry objects without specifying a key

Once we do the `name.startswith` and `is_dir` checks, do we even need the entries to be `os.DirEntry` objects anymore? In other words, could we do something like this?

```python
subject_dirs = sorted([d.name for d in os.scandir(path_data)
                       if d.name.startswith(subject_prefix)
                       and d.is_dir()])
```

Reply:
And in fact, you could even save yourself some path concatenation later by using `d.path` instead of `d.name`, so:
```py3
subject_dirs = sorted(d.path for d in os.scandir(path_data)
                      if d.name.startswith(subject_prefix)
                      and d.is_dir())
```

Reply:
Oh, except that the function itself returns relative paths rather than absolute paths, so using `d.path` doesn't work, sorry!